cmd/lakefs/cmd/config.go

```diff
@@ -0,0 +1,220 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/manifoldco/promptui"
+	"github.com/mitchellh/go-homedir"
+
+	"io/ioutil"
+
+	"encoding/json"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+// configuration model
+
+type Database struct {
+	ConnectionString string `yaml:"connection_string"`
+}
+
+type Gateways struct {
+	S3 struct {
+		DomainName string `yaml:"domain_name"`
+		Region     string `yaml:"region"`
+	} `yaml:"s3"`
+}
+
+type Configuration struct {
+	DB         Database               `yaml:"database"`
+	Blockstore map[string]interface{} `yaml:"blockstore"`
+	GW         Gateways               `yaml:"gateways"`
+}
+
+// configCmd represents the config command
+var configCmd = &cobra.Command{
+	Use:   "config",
+	Short: "User Interactive Config File generator",
+	Long:  `long description here...`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		var config Configuration
+
+		//loop for menu based interaction
+
+		var isEdit string
+		isEdit = "Yes"
+
+		for {
+
+			//Database
+			//connection string
+			prompt := promptui.Prompt{
+				Label:    "Database->connection string",
+				Validate: isEmptyInput,
+			}
+
+			connectionString, _ := prompt.Run()
+			config.DB.ConnectionString = connectionString
+
+			//Blockstore
+			config.Blockstore = make(map[string]interface{})
+			prompt = promptui.Prompt{
+				Label:    "Blockstore->Type",
```

Comment:
```suggestion
				Label:    "Blockstore type [s3|gs|local|mem|transient]",
```
Also validate that the value is one of those